pydantic_ai_slim/pydantic_ai/profiles/cerebras.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations as _annotations
+
+import warnings
+
+from pydantic_ai.exceptions import UserError
+
+from ._json_schema import JsonSchema, JsonSchemaTransformer
+
+
+class CerebrasJsonSchemaTransformer(JsonSchemaTransformer):
+    """Transforms the JSON Schema from Pydantic to be suitable for Cerebras.
+
+    Cerebras supports a subset of OpenAI's structured output capabilities, which is documented here:
+    - https://inference-docs.cerebras.ai/capabilities/structured-outputs#advanced-schema-features
+    - https://inference-docs.cerebras.ai/capabilities/structured-outputs#variations-from-openai's-structured-output-capabilities
+    - https://inference-docs.cerebras.ai/capabilities/tool-use
+
+    TODO: `transform` method is based on GoogleJsonSchemaTransformer, and it doesn't handle all cases mentioned in links above.
+    """
+
+    def __init__(self, schema: JsonSchema, *, strict: bool | None = None):
+        super().__init__(schema, strict=strict, prefer_inlined_defs=True, simplify_nullable_unions=False)
+
+    def transform(self, schema: JsonSchema) -> JsonSchema:
```

Comment:
Can we please reduce the duplication here with the `GoogleJsonSchemaTransformer`, perhaps by subclassing it?